### PR TITLE
Refs #29983 -- Added tests for FileBasedCache pathlib support.

### PR DESCRIPTION
**Pull request by jdufresne**

### CODE REVIEW

File: tests/cache/tests.py [hunk 1]
```diff
@@ -1473,6 +1474,14 @@ def test_empty_cache_file_considered_expired(self):
             self.assertIs(cache._is_expired(fh), True)
 
 
+class FileBasedCachePathLibTests(FileBasedCacheTests):
+    def setUp(self):
+        super().setUp()
+        for cache_params in settings.CACHES.values():
+            cache_params.update({'LOCATION': Path(self.dirname)})
+        setting_changed.send(self.__class__, setting='CACHES', enter=False)
```
Comment (pope1ni) [inline tests/cache/tests.py]: Do the changes to the settings need to be reverted in `tearDown()`?

File: tests/cache/tests.py [hunk 2]
```diff
@@ -1473,6 +1474,14 @@ def test_empty_cache_file_considered_expired(self):
             self.assertIs(cache._is_expired(fh), True)
 
 
+class FileBasedCachePathLibTests(FileBasedCacheTests):
+    def setUp(self):
+        super().setUp()
+        for cache_params in settings.CACHES.values():
+            cache_params.update({'LOCATION': Path(self.dirname)})
```
Comment (pope1ni) [inline tests/cache/tests.py]:
```python
cache_params['LOCATION'] = Path(self.dirname)
```

File: tests/cache/tests.py (hunk 1 above)
Comment (jdufresne) [reply tests/cache/tests.py]: Nope. This is already handled by the `override_settings` on the parent class.

File: tests/cache/tests.py (hunk 2 above)
Comment (jdufresne) [reply tests/cache/tests.py]: Good call. Fixed in the latest revision.

**Pull request merged**